homeassistant/components/philips_js/light.py

```diff
@@ -327,35 +344,39 @@ async def async_turn_on(self, **kwargs) -> None:
         """Turn the bulb on."""
         brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness)
         hs_color = kwargs.get(ATTR_HS_COLOR, self.hs_color)
-        effect = kwargs.get(ATTR_EFFECT, self.effect)
+        attr_effect = kwargs.get(ATTR_EFFECT, self.effect)
 
         if not self._tv.on:
             raise Exception("TV is not available")
 
-        mode, style, setting = _parse_effect(effect)
+        effect = AmbilightEffect.from_str(attr_effect)
+
+        if effect.style == "OFF":
```

Comment:
If the effect was asked to be set to off, seems it should go off? Why this extra code?

Reply:
Err. Never mind, we fall back to self.effect. Anyway it seem we handle this later with the is_on check?

Reply:
Yes you were right, there were two code paths now for checking if the light was one. Which made it quite confusing, unreadable and unmaintainable.

I have modified the `is_on` method now to return early when style is `OFF`. Than the ambilight can always be considered off.
Next I have replace the old logic with the new logic (restoring last selected effect or when unknown setting follow_video standard).

Tested for a bit and everything seems to work fine on my TV.